src/main/java/com/minecraftabnormals/abnormals_core/core/events/listener/ACEvents.java

```diff
@@ -59,4 +73,26 @@ public static void onUpdateEntity(LivingEvent.LivingUpdateEvent event) {
 			if (entity.getPersistentData().getBoolean(POISON_TAG)) ((IAgeableEntity) entity).resetGrowthProgress();
 		}
 	}
+
+	@SubscribeEvent(priority = EventPriority.LOWEST)
+	public static void onNoteBlockPlay(NoteBlockEvent.Play event) {
+		BlockPos pos = event.getPos();
+		BlockState state = event.getWorld().getBlockState(pos.relative(Direction.DOWN));
+		for (DataUtil.CustomNoteBlockInstrument instrument : SORTED_CUSTOM_NOTE_BLOCK_INSTRUMENTS) {
+			if (instrument.test(state)) {
+				SoundEvent sound = instrument.sound();
+				int note = event.getVanillaNoteId();
+				float f = (float) Math.pow(2.0D, (double) (note - 12) / 12.0D);
+				event.getWorld().playSound(null, pos, sound, SoundCategory.RECORDS, 3.0F, f);
+				if (!event.getWorld().isClientSide()) {
+					ResourceLocation noteKey = ForgeRegistries.PARTICLE_TYPES.getKey(ParticleTypes.NOTE);
+					if (noteKey != null)
+						NetworkUtil.spawnParticle(noteKey.toString(), pos.getX() + 0.5D, pos.getY() + 1.2D, pos.getZ() + 0.5D, (double) note / 24.0D, 0.0D, 0.0D);
+				}
```

Comment:
Is there a reason why `world.addParticle()` isn't used here instead?

Reply:
`NoteBlockEvent.Play` isn't called on the client.